### ISSUE TITLE
NameEmail validation fails when the name has a period in it

### ISSUE THREAD

**Issue by glynos** [pydantic/pydantic#11056]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

Using the NameEmail field type, I want to validate a name and email of the form ```Homer J. Simpson <homer@thesimpsons.com>```

However, when I try to validate this, it fails because Pydantic doesn't like the use of the period character to abbreviate the middle name. I would expect this to be valid.

### Example Code

```Python
from pydantic import BaseModel, NameEmail, ValidationError


class User(BaseModel):
    name_email: NameEmail

# This works
try:
    name_email = NameEmail(
        name="Homer J. Simpson",
        email="homer@thesimpsons.com",
    )
    print(f"{name_email}")
except ValidationError as e:
    print(f"{e}")

# This is a validation error
try:
    name_email = User.model_validate(
        {
            "name_email": "Homer J. Simpson <homer@thesimpsons.com>",
        }
    )
    print(f"{name_email}")
except ValidationError as e:
    print(f"{e}")
```

### Python, Pydantic & OS Version

```Text
pydantic version: 2.10.3
        pydantic-core version: 2.27.1
          pydantic-core build: profile=release pgo=false
                 install path: /Users/glynm/.pyenv/versions/3.12.7/envs/mantis-git/lib/python3.12/site-packages/pydantic
               python version: 3.12.7 (main, Oct 17 2024, 10:06:07) [Clang 16.0.0 (clang-1600.0.26.3)]
                     platform: macOS-15.1.1-arm64-arm-64bit
             related packages: mypy-1.13.0 typing_extensions-4.12.2
                       commit: unknown
```

**Comment by sangameshbagewadi** [comment]
The issue you're facing with Pydantic 2.10.3 is related to a change in behavior for fields marked as Optional or Maybe types when the default value is set to None.

In Pydantic 2.x, a stricter validation mechanism was introduced. By default, when you use Field(None), it implies that the field can be None, but the presence of a None value can cause issues when validation is done. The key difference here is that Field(None) now treats the field as required, which is why you're encountering the Field required error.


Solution example
`from typing import Optional
import uvicorn
from fastapi import FastAPI
from pydantic import BaseModel, Field

app = FastAPI()

class Item(BaseModel):
    field: Optional[int] = None  # Use Optional[int] and set default to None

@app.post("/items/")
async def create_item(item: Item):
    return item

uvicorn.run(app)
`

To solve the issue and allow the field to be blank or None without raising the validation error, you can make the following adjustment:

Use Field(default=None) instead of Field(None) to explicitly mark the field as optional without requiring a value.
You can also use Optional from typing instead of relying on Maybe, since Optional explicitly means the field can either have a value or be None

**Comment by glynos** [comment]
> The issue you're facing with Pydantic 2.10.3 is related to a change in behavior for fields marked as Optional or Maybe types when the default value is set to None.
> 
> In Pydantic 2.x, a stricter validation mechanism was introduced. By default, when you use Field(None), it implies that the field can be None, but the presence of a None value can cause issues when validation is done. The key difference here is that Field(None) now treats the field as required, which is why you're encountering the Field required error.
> 
> Solution example `from typing import Optional import uvicorn from fastapi import FastAPI from pydantic import BaseModel, Field
> 
> app = FastAPI()
> 
> class Item(BaseModel): field: Optional[int] = None # Use Optional[int] and set default to None
> 
> @app.post("/items/") async def create_item(item: Item): return item
> 
> uvicorn.run(app) `
> 
> To solve the issue and allow the field to be blank or None without raising the validation error, you can make the following adjustment:
> 
> Use Field(default=None) instead of Field(None) to explicitly mark the field as optional without requiring a value. You can also use Optional from typing instead of relying on Maybe, since Optional explicitly means the field can either have a value or be None

This has nothing to do with optional values, it's related to a regex in the email validator which explicitly excludes the period character in the display name for NameEmail.

**Comment by sangameshbagewadi** [comment]
You're absolutely correct — the issue lies in the NameEmail regex validator's strict enforcement of rules for the display name, which excludes the use of a period (.) in certain contexts. Pydantic's NameEmail relies on a strict interpretation of the RFC 5322 standard for email addresses, and its display name validation doesn't account for names with periods used as abbreviations.
To resolve this, you need to override the default behavior of NameEmail and adjust the regex used for validating the display name.

**Comment by mpkocher** [comment]
Types are implemented in different styles within Pydantic. 

The validation only really works in the context of BaseModel, or TypeAdapter. I suspect this is the source of the inconsistency in the two (stylistically) different examples you provided. This is perhaps creating some confusion.

More info on TypeAdapter. 

https://docs.pydantic.dev/latest/api/type_adapter/


To demonstrate this in the REPL.

(No validation is occurring here, only assigning of values)

```bash
In [12]: from pydantic import NameEmail, PositiveInt

In [13]: PositiveInt(-7)
Out[13]: -7

In [14]: NameEmail(1, 2)
Out[14]: NameEmail(name=1, email=2)

In [15]: NameEmail(PositiveInt(-7), NameEmail("a", "b"))
Out[15]: NameEmail(name=-7, email=NameEmail(name='a', email='b'))
```

I believe the underlying issue can be resolved by quoting the string. 

```bash
In [19]: name_email = TypeAdapter(NameEmail).validate_python('"Homer J. Simpson" <jsimpson@a.com>')

In [20]: name_email
Out[20]: NameEmail(name='Homer J. Simpson', email='jsimpson@a.com')
```

**Comment by glynos** [comment]
Thanks for the responses, but perhaps my original example wasn't clear. I don't have an issue with creating a NameEmail with or without a TypeAdapter, but the issue is when I want to validate it from an input string. I don't control the input, so quoting the display name as suggested is not an option. To adapt my original example, you can see there are two different outcomes when I would expect them to behave the same:

```
rom pydantic import BaseModel, NameEmail, ValidationError


class User(BaseModel):
    name_email: NameEmail


# This works
try:
    user = User(
        name_email=NameEmail(
           name="Homer J. Simpson",
            email="homer@thesimpsons.com",
        ),
    )
    print(f"{user}")
except ValidationError as e:
    print(f"{e}")

# This is a validation error
try:
    user = User.model_validate(
        {
            "name_email": "Homer J. Simpson <homer@thesimpsons.com>",
        }
    )
    print(f"{user}")
except ValidationError as e:
    print(f"{e}")
```

To respond to @sangameshbagewadi , section 4.1 in the RFC document you cite explicitly states that names with periods in them must be interpreted properly: https://datatracker.ietf.org/doc/html/rfc5322#section-4.1

**Comment by sangameshbagewadi** [comment]
Try this


`from pydantic import BaseModel, ValidationError
from pydantic.networks import EmailStr
from typing import Any


class CustomNameEmail:
    def __init__(self, name: str, email: str):
        self.name = name
        self.email = email

    @classmethod
    def validate(cls, value: Any) -> "CustomNameEmail":
        if isinstance(value, str):
            try:
                # Split the input into "name" and "email" parts
                name, email = value.rsplit(" <", 1)
                email = email.rstrip(">")
                if not email or "@" not in email:
                    raise ValueError("Invalid email format")
                # Return a validated CustomNameEmail instance
                return cls(name=name.strip(), email=email.strip())
            except ValueError:
                raise ValueError(f"Invalid NameEmail format: {value}")
        raise TypeError("Expected a string for NameEmail")

    def __repr__(self):
        return f"{self.name} <{self.email}>"

    def __str__(self):
        return self.__repr__()


class User(BaseModel):
    name_email: CustomNameEmail

    @classmethod
    def model_validate(cls, value: dict):
        # Manually apply validation using CustomNameEmail
        try:
            value['name_email'] = CustomNameEmail.validate(value['name_email'])
        except (ValueError, TypeError) as e:
            raise ValidationError(str(e))
        return cls(**value)


# Example Usage

# This works
try:
    user = User(
        name_email=CustomNameEmail(
            name="Homer J. Simpson",
            email="homer@thesimpsons.com",
        )
    )
    print(f"User (manual init): {user}")
except ValidationError as e:
    print(f"Validation Error: {e}")

# This is now consistent
try:
    user = User.model_validate(
        {
            "name_email": "Homer J. Simpson <homer@thesimpsons.com>",
        }
    )
    print(f"User (string input): {user}")
except ValidationError as e:
    print(f"Validation Error: {e}")
`

**Comment by Viicos** [comment]
@sangameshbagewadi we appreciate the help but your (AI generated?) answers are irrelevant and poorly formatted. I marked them as off topic, please avoid doing so again.

**Comment by Viicos** [comment]
The `NameEmail` type is a bit confusing, because it is meant to be used as an annotation in a Pydantic context (i.e. as a field type/with `TypeAdapter`).

However, it is instantiable even though no validation is performed on instantiation.

Doing:

```python
user = User(
    name_email=NameEmail(
        name="Homer J. Simpson",
        email="homer@thesimpsons.com",
    ),
)
```

will *not* perform any validation, because the validation function is defined as:

https://github.com/pydantic/pydantic/blob/32f405bcf6171602ffe754f1fca1681c5ddee96e/pydantic/networks.py#L1008-L1013

which is arguably flawed: the `input_value` type hint indicates that validating as a `NameInstance` is supported (because we allow `Self`) but we don't perform any validation here (probably it was assumed here that if it is an instance, it was already validated, perhaps on init or from another model validation).

cc @sydney-runkle, this is similar to the URL types, we are missing consistency here.

Whether we should allow period in names or not, from my understanding of https://datatracker.ietf.org/doc/html/rfc5322#section-4 we should. It is an obsolete token, but:

> they MUST be accepted and parsed by a conformant receiver.

**Comment by sangameshbagewadi** [comment]
> @sangameshbagewadi we appreciate the help but your (AI generated?) answers are irrelevant and poorly formatted. I marked them as off topic, please avoid doing so again.

Thank you for feedback but using AI in search for possible solutions is not out of the question  it is using tools of tomorrow. It is not a response that failed but the issue itself, for its subtlety cannot be aptly conveyed in that media. Instead of discarding an idea, it is better to offer constructive suggestions.

**Comment by Viicos** [comment]
Thanks but no thanks. Please take time to assess the really poor quality of your answers, especially [this one](https://github.com/pydantic/pydantic/issues/11056#issuecomment-2523096948).

**Comment by sydney-runkle** [comment]
@Viicos indeed, the inconsistency here makes my head hurt. Let's chat about this at our next sync :(.

**Comment by SahilNambiar11** [comment]
please review
